### ISSUE TITLE
Plugins can't load data from the database

### ISSUE THREAD

**Issue by halworsen** [rebel1324/NutScript#177]
See 4973f34 and ![this line in particular](https://github.com/rebel1324/NutScript/blob/7372c6809e45f44c212edf6118d020309891103b/gamemode/init.lua#L20)

This causes issues with plugins that loads data from the database because the database loads on the first think/tick of the game, which isn't going to run before a player joins. As of 4973f34 all plugins load data immediately, so the end result is that plugins load data before the database initializes. It worked fine before because the `LoadData` hook was called in a timer as well, so both waited until a player joined the game, with the database loading first.

TL:DR; Plugins load data before the database is initialized. Every time.

**Comment by rebel1324** [comment]
So do you need something like PreLoadData?

**Comment by halworsen** [comment]
No, the plugin data loading needs to happen at a later tick. The problem is that the load order is like this:
Tick "0" (server startup): Plugin data is loaded
Tick 1 (only happens after a player loads): Database is loaded

The loading order needs to be the other way around. The database is being loaded in tick 1 because of the `timer.Simple(0, ...` in init.lua

**Comment by rebel1324** [comment]
Oh yeah maybe I need to do something like OnDatabaseLoaded

**Comment by halworsen** [comment]
That could work, yeah

**Comment by Rubberbandits** [comment]
Fixed by https://github.com/rebel1324/NutScript/pull/181
